independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/observers/inheritance/typevariable/ObserverInheritanceTypeVariableTest.java

```diff
@@ -0,0 +1,85 @@
+package io.quarkus.arc.test.observers.inheritance.typevariable;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.Unremovable;
+import io.quarkus.arc.test.ArcTestContainer;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.event.Event;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+/**
+ * https://github.com/quarkusio/quarkus/issues/25364
+ */
+public class ObserverInheritanceTypeVariableTest {
+
+    @RegisterExtension
+    public ArcTestContainer container = new ArcTestContainer(MyEvent.class, MyAEvent.class, MyBEvent.class, MyAService.class,
+            MyBService.class, EventSource.class);
+
+    @Test
+    public void testNotification() {
+        Arc.container().instance(EventSource.class).get().sendA();
+        assertNotNull(MyAService.event);
+        assertNull(MyBService.event);
+    }
+
+    static class MyEvent {
+    }
+
+    static class MyAEvent extends MyEvent {
+    }
+
+    static class MyBEvent extends MyEvent {
+    }
+
+    static abstract class AbstractService<E extends MyEvent> {
+
+        void onEvent(@Observes E myEvent) {
```

Comment:
Should the test cover the case of asynchronous observer methods as well?

Reply:
I don't think it's necessary. Both sync and async observers are created in the same way.